src/doc/reference.md

```diff
@@ -1325,10 +1321,44 @@ trait Shape {
 }
 ```
 
-This defines a trait with two methods. All values that have
-[implementations](#implementations) of this trait in scope can have their
-`draw` and `bounding_box` methods called, using `value.bounding_box()`
-[syntax](#method-call-expressions).
+This defines a trait with two methods, `draw` and `bounding_box`. These
+methods can be invoked on any value whose type has an
+[implementation](#implementations) of this trait. The [method call
+syntax](#method-call-expressions) looks like this: `value.bounding_box()`.
+
+Trait methods can be defined to take a [`self` type](#self-types) as a
+parameter. In the example above, each method takes `&self` as its first
+parameter. This is sometimes called the _receiver_ of the method, and indicates
+that for any type `T` implementing `Shape`, invoking one of the `Shape` methods
+will immutably borrow the `T` value for the duration of the method call.
+
+Traits can include default implementations of methods, as in:
+
+```
+trait Foo {
+    fn bar(&self);
+
+    fn baz(&self) { println!("We called baz."); }
+}
+```
+
+Here the `baz` method has a default implementation, so types that implement
+`Foo` need only implement `bar`. It is also possible for implementing types
+to override a method that has a default implementation.
+
+It is not required that trait methods use a `self` type as a parameter. Methods
+that lack `self` types are called _static methods_. A common use for static
```

Comment:
I think we're starting to call these "associated functions" instead of static methods to stay in line with associated constants/items